### PR TITLE
PP-5003: Add fee table schema migration

### PR DESCRIPTION
**Pull request by sfount**
* Adds initial columns required for modelling fees
* Indexing gateway processing details will follow when we know what keys
are required

Fees are modelled separately to reflect the second account debit transaction that is required on the Stripe gateway. For this use case they are 1:1 for a charge which is enforced in https://github.com/alphagov/pay-connector/pull/1161/commits/87c575f81a9dffc417c4c30b721442f282ea9b67. If multiple fees for a given charge become a use case in the future this constraint can be updated along with the "view" code for charges and their fees.

### CODE REVIEW

File: src/main/resources/migrations.xml [hunk 1]
```diff
@@ -1013,5 +1013,33 @@
...
+            <column name="external_id" type="char(26)" />
+            <column name="charge_id" type="bigserial">
+                <constraints foreignKeyName="fk__charge_fees"
+                             referencedTableName="charges"
+                             referencedColumnNames="id"
+                             nullable="false" />
+            </column>
+            <column name="fee_due" type="bigint" />
+            <column name="fee_collected" type="bigint" />
+            <column name="created" type="timestamp without timezone" />
```
Comment (mrlumbu) [inline src/main/resources/migrations.xml]: can this be `created_date` for consistency

File: src/main/resources/migrations.xml [hunk 2]
```diff
@@ -1013,5 +1013,33 @@
...
+            <column name="gateway_transaction_id"  type="text">
+                <constraints nullable="true" />
+            </column>
+            <column name="gateway_processing_details" type="jsonb" />
+        </createTable>
+    </changeSet>
+    
+    <changeSet id="index gateway_transaction_id for fee table" author="">
+        <createIndex tableName="fees" indexName="itx_fees_gateway_transaction_id">
```
Comment (mrlumbu) [inline src/main/resources/migrations.xml]: `s/itx/idx/` ?

**Review by mrlumbu — CHANGES_REQUESTED** [review]

File: src/main/resources/migrations.xml (hunk 1 above)
Comment (sfount) [reply src/main/resources/migrations.xml]: 👍 will line this up with other entities

File: src/main/resources/migrations.xml [hunk 3]
```diff
@@ -1013,5 +1013,33 @@
...
+            </column>
+            <column name="gateway_processing_details" type="jsonb" />
+        </createTable>
+    </changeSet>
+    
+    <changeSet id="index gateway_transaction_id for fee table" author="">
+        <createIndex tableName="fees" indexName="idx_fees_gateway_transaction_id">
+            <column name="gateway_transaction_id"></column>
+        </createIndex>
+    </changeSet>
```
Comment (mrlumbu) [inline src/main/resources/migrations.xml:1057]: probs need index on charge_id and external_id

Comment (sfount) [reply src/main/resources/migrations.xml:1057]: I've added these indexes to the same `changeSet`

**Comment by sfount** [comment]
EDIT Update fee constraints given: 
> Fees are modelled separately to reflect the second account debit transaction that is required on the Stripe gateway. For this use case they are 1:1 for a charge which is enforced in 87c575f. If multiple fees for a given charge become a use case in the future this constraint can be updated along with the "view" code for charges and their fees.

**Review by mrlumbu — APPROVED** [review]

**Pull request merged**